uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-actions.component.js

```diff
@@ -0,0 +1,224 @@
+  'common/services/hr-settings'
+], function (_, moment, components) {
+  components.component('leaveRequestActions', {
+    bindings: {
+      leaveRequest: '<',
+      leaveRequestStatuses: '<',
+      absenceTypes: '<',
+      /**
+       * Role is not a permission level in this case.
+       * For example Manager can act as Staff
+       * and Admin can act as either Manager or Staff.
+       */
+      role: '<'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-request-actions.html';
+    }],
+    controllerAs: 'actions',
+    controller: ['$log', '$q', '$rootScope', 'OptionGroup', 'dialog', 'shared-settings', controller]
+  });
+
+  function controller ($log, $q, $rootScope, OptionGroup, dialog, sharedSettings) {
+    $log.debug('Component: leave-request-action-dropdown');
+
+    var vm = this;
+    var actions = {
+      edit: {
+        label: 'Edit',
+        statuses: [sharedSettings.statusNames.awaitingApproval]
+      },
+      respond: {
+        label: 'Respond',
+        statuses: [sharedSettings.statusNames.moreInformationRequired]
+      },
+      view: {
+        label: 'View',
+        statuses: [
+          sharedSettings.statusNames.approved,
+          sharedSettings.statusNames.rejected,
+          sharedSettings.statusNames.cancelled
+        ]
+      },
+      approve: {
+        label: 'Approve',
+        isDirectAction: true,
+        statuses: [sharedSettings.statusNames.awaitingApproval],
+        dialog: {
+          title: 'Approval',
+          btnClass: 'success',
+          btnLabel: 'Approve',
+          msg: 'Please confirm approval'
+        }
+      },
+      reject: {
+        label: 'Reject',
+        isDirectAction: true,
+        statuses: [sharedSettings.statusNames.awaitingApproval],
+        dialog: {
+          title: 'Rejection',
+          btnClass: 'warning',
+          btnLabel: 'Reject',
+          msg: 'Please confirm rejection'
+        }
+      },
+      cancel: {
+        label: 'Cancel',
+        isDirectAction: true,
+        statuses: [
+          sharedSettings.statusNames.awaitingApproval,
+          sharedSettings.statusNames.moreInformationRequired
+        ],
+        dialog: {
+          title: 'Cancellation',
+          btnClass: 'danger',
+          btnLabel: 'Confirm',
+          msg: 'Please confirm cancellation'
+        }
+      },
+      delete: {
+        label: 'Delete',
+        isDirectAction: true,
+        statuses: [
+          sharedSettings.statusNames.awaitingApproval,
+          sharedSettings.statusNames.moreInformationRequired,
+          sharedSettings.statusNames.approved,
+          sharedSettings.statusNames.rejected,
+          sharedSettings.statusNames.cancelled
+        ],
+        dialog: {
+          title: 'Deletion',
+          btnClass: 'danger',
+          btnLabel: 'Confirm',
+          msg: 'This cannot be undone'
+        }
+      }
+    };
+
+    vm.list = [];
```

Comment:
again give descriptive names, `list` of what? isn't `allowedActions` better? The name is more descriptive and it implies that it's a list